src/Api/HttpApi.php

```diff
@@ -44,13 +43,13 @@ abstract class HttpApi
      */
     protected $requestBuilder;
 
-    public function __construct($httpClient, RequestBuilder $requestBuilder, Hydrator $hydrator)
+    /**
+     * @param ClientInterface $httpClient     Client used to send the requests
+     * @param RequestBuilder  $requestBuilder Builder that creates request based on params
+     * @param Hydrator        $hydrator       Transforms request body into array
```

Comment:
Since the types are already defined in the code, it seems redundant to repeat them in the docs here if we aren't going to make them more specific. I think the following would work:
```suggestion
     * @param $httpClient     Client used to send the requests
     * @param $requestBuilder Builder that creates request based on params
     * @param $hydrator       Transforms request body into array
```
(And ditto for the other doc-blocks in this PR.)

Reply:
The `@param TypeHint $variableName` is redundant, but it also breaks the PSR, as far as I can remember. When adding the `@param` we should also add a description to explain what that variable is.

Consider something like `@param GenericRepositoryInterface $repo` - It's redundant, but you might want to describe what you expect from that repo to be passed.

In my opinion, keeping it is not problematic at all.

Reply:
I looked it up, and you're right: omitting the type from an `@param` does break standards, namely the [PSR-19 draft](https://github.com/php-fig/fig-standards/blob/master/proposed/phpdoc-tags.md#510-param). Thanks for pointing that out.